Code/Framework/AzCore/AzCore/Script/ScriptProperty.cpp

```diff
@@ -15,6 +15,49 @@
 
 namespace AZ
 {
+    // Add TypeInfo and RTTI Reflection within the cpp file
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptProperty, "ScriptProperty", "{D227D737-F1ED-4FB3-A1FB-38E4985D2E7A}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptProperty);
+
+    AZ_TYPE_INFO_WITH_NAME_IMPL(FunctionalScriptProperty, "FunctionalScriptProperty", "{57D7418D-6B14-4A02-B50E-2E409D23CFC6}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(FunctionalScriptProperty, ScriptProperty);
+
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyNil, "ScriptPropertyNil", "{ACAD23F6-5E75-460E-BD77-1B477750264F}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyNil, ScriptProperty);
+
+    // ScriptPropertyBoolean is serialized to a prefab file using the type name of AzFramwork::ScriptPropertyBoolean
+    // So make sure the old name is being used
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyBoolean, "AzFramework::ScriptPropertyBoolean", "{EA7335F8-5B9F-4744-B805-FEF9240451BD}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyBoolean, ScriptProperty);
+
+    // ScriptPropertyNumber is serialized to a prefab file using the type name of AzFramwork::ScriptPropertyNumber
+    // So make sure the old name is being used
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyNumber, "AzFramework::ScriptPropertyNumber", "{5BCDFDEB-A75D-4E83-BB74-C45299CB9826}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyNumber, ScriptProperty);
+
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyString, "ScriptPropertyString", "{A0229C6D-B010-47E7-8985-EE220FC7BFAF}");
```

Comment:
I believe this should have AzFramework:: in front of its name.
https://github.com/o3de/o3de/blob/152bc0a1851d881fe735adf54ec93e1ad7875b11/Code/Framework/AzCore/AzCore/Script/ScriptProperty.h#L220

Reply:
The types other than ScriptPropertyBoolean, ScriptPropertyNumber and ScriptPropertyGenericClass, are serialized to any JSON files at the moment, but I can prefix their names with `AzFramework::` to be safe.